File: core/src/main/java/com/pagoda/core/model/Person.java
package com.pagoda.core.model;

import java.time.LocalDate;

public record Person(
        String lastName,
        String firstName,
        Gender gender,
        LocalDate birthDate,
        String birthPlace,
        String permanentResidence,
        String temporaryResidence,
        String hometown,
        String nationality,
        String ethnicity,
        String dharmaName,
        LocalDate ordinationDate,
        String idNumber,
        LocalDate idIssueDate,
        String idIssuePlace,
        String email,
        String phoneNumber,
        String notes,
        PersonType type) {
}

File: web-app/src/types/person.ts
export interface PersonDTO {
  id?: string
  lastName: string
  firstName: string
  gender: string
  birthDate: string
  birthPlace: string
  permanentResidence: string
  temporaryResidence: string
  hometown: string
  nationality: string
  ethnicity: string
  dharmaName: string
  ordinationDate: string
  idNumber: string
  idIssueDate: string
  idIssuePlace: string
  email: string
  phoneNumber: string
  notes: string
  type: PersonType
}

export type PersonType = 'BUDDHIST' | 'LAY_BROTHER' | 'MASTER'
export type Gender = 'MALE' | 'FEMALE'
